docs/src/test-parallel-js.md

```diff
@@ -3,15 +3,27 @@ id: test-parallel
 title: "Parallelism and sharding"
 ---
 
-Playwright Test runs tests in parallel by default, using multiple worker processes.
+Playwright Test uses multiple worker processes to run tests in parallel. Each worker process creates a new environment to run tests. By default, Playwright Test reuses the worker as much as it can to make testing faster.
 
-<!-- TOC -->
+Should any test fail, Playwright will discard entire worker process along with the browsers used and will start a new one. That way failing tests can't affect healthy ones.
 
-## Workers
+* By default, Playwright test [runs test files in parallel](#running-test-files-in-parallel).
```

Comment:
. Tests from the same file will run sequentially on one and the same worker.

Or just inline linked section here otherwise it's hard to read.